eZ/Bundle/EzPublishCoreBundle/Command/CleanupVersionsCommand.php

```diff
@@ -0,0 +1,266 @@
+
+use Doctrine\DBAL\Connection;
+use Exception;
+use eZ\Bundle\EzPublishCoreBundle\ApiLoader\RepositoryConfigurationProvider;
+use eZ\Publish\API\Repository\Repository;
+use eZ\Publish\API\Repository\Values\Content\VersionInfo;
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentException;
+use PDO;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\Output;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class CleanupVersionsCommand extends Command
+{
+    const DEFAULT_REPOSITORY_USER = 'admin';
+
+    const VERSION_DRAFT = 'draft';
+    const VERSION_ARCHIVED = 'archived';
+    const VERSION_PUBLISHED = 'published';
+    const VERSION_ALL = 'all';
+
+    const VERSION_STATUS = [
+        self::VERSION_DRAFT => VersionInfo::STATUS_DRAFT,
+        self::VERSION_ARCHIVED => VersionInfo::STATUS_ARCHIVED,
+        self::VERSION_PUBLISHED => VersionInfo::STATUS_PUBLISHED,
+    ];
+
+    /**
+     * @var \eZ\Publish\API\Repository\Repository
+     */
+    private $repository;
+
+    /**
+     * @var \eZ\Publish\API\Repository\UserService
+     */
+    private $userService;
+
+    /**
+     * @var \eZ\Publish\API\Repository\ContentService
+     */
+    private $contentService;
+
+    /**
+     * @var \eZ\Publish\API\Repository\PermissionResolver
+     */
+    private $permissionResolver;
+
+    /**
+     * @var \eZ\Bundle\EzPublishCoreBundle\ApiLoader\RepositoryConfigurationProvider
+     */
+    private $repositoryConfigurationProvider;
+
+    /**
+     * @var \Doctrine\DBAL\Driver\Connection
+     */
+    private $connection;
+
+    public function __construct(
+        Repository $repository,
+        RepositoryConfigurationProvider $repositoryConfigurationProvider,
+        Connection $connection
+    ) {
+        $this->repository = $repository;
+        $this->repositoryConfigurationProvider = $repositoryConfigurationProvider;
+        $this->connection = $connection;
+
+        parent::__construct();
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        parent::initialize($input, $output);
+
+        $this->userService = $this->repository->getUserService();
+        $this->contentService = $this->repository->getContentService();
+        $this->permissionResolver = $this->repository->getPermissionResolver();
+
+        $this->permissionResolver->setCurrentUserReference(
+            $this->userService->loadUserByLogin($input->getOption('user'))
+        );
+    }
+
+    protected function configure()
+    {
+        $config = $this->repositoryConfigurationProvider->getRepositoryConfig();
+
+        $this
+            ->setName('ezplatform:content:cleanup-versions')
+            ->setDescription('Remove unwanted content versions. It keeps published version untouched. By default, it keeps also the last archived/draft version.')
```

Comment:
Nipick: 'unwanted' is not the best term to use here: it is not descriptive of what it is considered unwanted